src/openrct2/entity/Guest.cpp

```diff
@@ -3627,9 +3627,12 @@ void Guest::UpdateRideLeaveEntranceWaypoints(const Ride& ride)
     CoordsXY waypoint = rtd.GetGuestWaypointLocation(*vehicle, ride, CurrentRideStation);
 
     const auto waypointIndex = Var37 / 4;
```

Comment:
pretty sure the upper part is a rotation.

Reply:
Yeah theres another variable masking it with `& 3`